src/meta-srv/src/metasrv/builder.rs

```diff
@@ -237,6 +244,8 @@ impl MetaSrvBuilder {
             ddl_manager,
             table_metadata_manager,
             greptimedb_telemerty_task: get_greptimedb_telemetry_task(meta_peer_client).await,
+            publish: Some(publish),
+            subscribe_manager: Some(sub_manager),
```

Comment:
If the two components `publish` and `subscribe_manager` must appear in pairs, that is, either all are `Some` or all are `None`, then I suggest wrapping the two components together.